### DISCUSSION TITLE
ipconfig: no devices to configure

### DISCUSSION THREAD

**Discussion by nek009** [ltsp/ltsp#229] [Q&A]
I tries to use surface go as LTSP client, but it stopped a half way of the sequence of start-up as like pic.
I confirmed that desktop PCs which uses legacy bios works.

![surfacegoerror](https://user-images.githubusercontent.com/6490702/72197434-93d7f880-3464-11ea-9cd1-2d5730e94a15.jpg)

**Comment by alkisg** [comment]
Hi,

the screenshot is a thumbnail (very small) so I can't see anything in it.

Based on what you said in the #ltsp IRC channel, I believe your initramfs is lacking the driver for your Ethernet.

Can you boot your surface with an Ubuntu 18.04 live USB stick and run these commands and paste the output?

```shell
lspci -nn -k | grep -A 3 Ethernet
lsmod
```

**Comment by nek009** [comment]
Thanks for reply.
The outputs are as follws:

lspci -nn -k | grep -A 3 Ethernet
(none, so donot recognize it)

lsmod

Module                  Size  Used by
rfcomm                 77824  4
arc4                   16384  2
intel_rapl             20480  0
x86_pkg_temp_thermal    16384  0
intel_powerclamp       16384  0
coretemp               16384  0
snd_soc_skl           102400  0
kvm_intel             208896  0
ath10k_pci             40960  0
snd_soc_skl_ipc        61440  1 snd_soc_skl
snd_soc_sst_ipc        16384  1 snd_soc_skl_ipc
snd_soc_sst_dsp        32768  1 snd_soc_skl_ipc
ath10k_core           417792  1 ath10k_pci
snd_hda_ext_core       24576  1 snd_soc_skl
cmac                   16384  1
snd_hda_codec_hdmi     49152  1
snd_soc_acpi           16384  1 snd_soc_skl
bnep                   20480  2
kvm                   626688  1 kvm_intel
snd_soc_core          229376  1 snd_soc_skl
snd_hda_codec_realtek   106496  1
snd_compress           20480  1 snd_soc_core
ac97_bus               16384  1 snd_soc_core
snd_hda_codec_generic    73728  1 snd_hda_codec_realtek
snd_pcm_dmaengine      16384  1 snd_soc_core
ath                    32768  1 ath10k_core
snd_hda_intel          40960  3
mac80211              802816  1 ath10k_core
snd_hda_codec         126976  4 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec_realtek
irqbypass              16384  1 kvm
crct10dif_pclmul       16384  0
crc32_pclmul           16384  0
snd_hda_core           81920  7 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_ext_core,snd_hda_codec,snd_hda_codec_realtek,snd_soc_skl
ghash_clmulni_intel    16384  0
pcbc                   16384  0
btusb                  45056  0
snd_hwdep              20480  1 snd_hda_codec
snd_pcm                98304  8 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_ext_core,snd_hda_codec,snd_soc_core,snd_soc_skl,snd_hda_core,snd_pcm_dmaengine
aesni_intel           200704  2
btrtl                  16384  1 btusb
snd_seq_midi           16384  0
btbcm                  16384  1 btusb
intel_rapl_perf        16384  0
serio_raw              16384  0
hid_sensor_accel_3d    16384  1
snd                    81920  19 snd_hda_codec_generic,snd_seq,snd_seq_device,snd_hda_codec_hdmi,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek,snd_timer,snd_compress,snd_soc_core,snd_pcm,snd_rawmidi
intel_wmi_thunderbolt    16384  0
hid_sensor_gyro_3d     16384  0
soundcore              16384  1 snd
hid_sensor_als         16384  1
hid_sensor_rotation    16384  0
ipu3_cio2              32768  0
hid_sensor_trigger     16384  9 hid_sensor_gyro_3d,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_rotation
ucsi_acpi              16384  0
industrialio_triggered_buffer    16384  4 hid_sensor_gyro_3d,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_rotation
idma64                 20480  0
v4l2_fwnode            20480  1 ipu3_cio2
kfifo_buf              16384  1 industrialio_triggered_buffer
virt_dma               16384  1 idma64
hid_sensor_iio_common    16384  5 hid_sensor_gyro_3d,hid_sensor_trigger,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_rotation
videobuf2_dma_sg       16384  1 ipu3_cio2
int3403_thermal        16384  0
processor_thermal_device    16384  0
industrialio           69632  9 industrialio_triggered_buffer,hid_sensor_gyro_3d,hid_sensor_trigger,kfifo_buf,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_rotation
typec_ucsi             32768  1 ucsi_acpi
int3400_thermal        16384  0
videobuf2_memops       16384  1 videobuf2_dma_sg
int340x_thermal_zone    16384  2 int3403_thermal,processor_thermal_device
acpi_thermal_rel       16384  1 int3400_thermal
intel_lpss_pci         20480  0
typec                  32768  1 typec_ucsi
videobuf2_v4l2         24576  1 ipu3_cio2
intel_lpss             16384  1 intel_lpss_pci
videobuf2_common       40960  2 ipu3_cio2,videobuf2_v4l2
soc_button_array       16384  0
mei_me                 40960  0
videodev              188416  4 v4l2_fwnode,ipu3_cio2,videobuf2_v4l2,videobuf2_common
intel_pch_thermal      16384  0
intel_soc_dts_iosf     16384  1 processor_thermal_device
intel_hid              16384  0
hid_generic            16384  0
usbhid                 49152  0
i915                 1740800  9
i2c_algo_bit           16384  1 i915
drm_kms_helper        172032  1 i915
syscopyarea            16384  1 drm_kms_helper
sysfillrect            16384  1 drm_kms_helper
sysimgblt              16384  1 drm_kms_helper
fb_sys_fops            16384  1 drm_kms_helper
nvme                   36864  0
rtsx_pci_sdmmc         24576  0
drm                   458752  5 drm_kms_helper,i915
nvme_core              77824  1 nvme
uas                    24576  0
intel_ish_ipc          20480  0
rtsx_pci               65536  2 rtsx_pci_sdmmc,rtsx_pci_ms
intel_ishtp            40960  2 intel_ishtp_hid,intel_ish_ipc
i2c_hid                20480  0
wmi                    24576  1 intel_wmi_thunderbolt
hid                   122880  6 i2c_hid,usbhid,hid_multitouch,hid_sensor_hub,intel_ishtp_hid,hid_generic
usb_storage            69632  2 uas
video                  45056  1 i915
i915                 1740800  9
i2c_algo_bit           16384  1 i915
drm_kms_helper        172032  1 i915
syscopyarea            16384  1 drm_kms_helper
sysfillrect            16384  1 drm_kms_helper
sysimgblt              16384  1 drm_kms_helper
fb_sys_fops            16384  1 drm_kms_helper
nvme                   36864  0
rtsx_pci_sdmmc         24576  0
drm                   458752  5 drm_kms_helper,i915
nvme_core              77824  1 nvme
uas                    24576  0
intel_ish_ipc          20480  0
rtsx_pci               65536  2 rtsx_pci_sdmmc,rtsx_pci_ms
intel_ishtp            40960  2 intel_ishtp_hid,intel_ish_ipc
i2c_hid                20480  0
wmi                    24576  1 intel_wmi_thunderbolt
hid                   122880  6 i2c_hid,usbhid,hid_multitouch,hid_sensor_hub,intel_ishtp_hid,hid_generic
usb_storage            69632  2 uas
video                  45056  1 i915
pinctrl_sunrisepoint    28672  0
pinctrl_intel          20480  1 pinctrl_sunrisepoint

**Comment by nek009** [comment]
Thanks for reply.
The output is as follow:

lspci -nn -k | grep -A 3 Ethernet:

(none)
ls mod

**Comment by alkisg** [comment]
A quick googling shows that Surface doesn't support Linux, it has no Ethernet drivers etc.
Since LTSP is build on top of Linux, Surface can't be used as an LTSP client.

There's an effort there to add drivers for Linux for surface; if you make Ethernet work, then you can also make LTSP work:
https://github.com/jakeday/linux-surface/issues/259

**Comment by nek009** [comment]
Really thanks. Surface Go works with advices on here and on IRC.
Perhaps, the following procedure includes unnecessary ones.

Machine:Suface Go and Microsoft USB Type-C Ethernet/USB adapter
Security boot on Surface Go turns off. 

1) Let Surface Go booting with ubuntu 18.04 live cd.
2) Get the module list by ismod on Surface Go.
3) Add the module list and 'cdc_ether' to /etc/initramfs-too/modules on LTSP server. ('cdc_ether' is not in the module list, and perhaps enough to add only 'cdc_ether'.)
4) update-initramfs -u
5) ltsp image /; ltsp ipxe; ltsp nfs; ltsp initrd;
